test/metabase/upload_test.clj

```diff
@@ -422,6 +443,38 @@
                       (t2/select-one Field :database_position 1 :table_id (:id table)))))
             (is (some? table))))))))
 
+(deftest load-from-csv-offset-datetime-test
+  (testing "Upload a CSV file with a datetime column"
+    (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
+      (mt/with-empty-db
+        (with-redefs [driver/db-default-timezone (constantly "Z")
+                      upload/current-database    (constantly (mt/db))]
```

Comment:
I think you could have avoided the redef here, which couples the test with the implementation
```
(tu/with-temporary-setting-values [uploads-database-id (:id (mt/db))]
```